src/preprocess/run_tg.py

```diff
@@ -43,105 +66,143 @@
     #           # scatter_plot_path=os.path.join(TG_OUT, 'global_pca.html'),
     #           bin_file_type='--bfile')
 
+
     # 1. Centralised variant QC
-    logger.info("Centralised variant QC")
     varqc_prefix = TG_BFILE_PATH + '_varqc'
-    QC.qc(input_prefix=TG_BFILE_PATH, output_prefix=varqc_prefix, qc_config=variant_qc_config)
+    if Stage.VARIANT_QC in stages:
+        logger.info('Centralised variant QC')
+        QC.qc(input_prefix=TG_BFILE_PATH, output_prefix=varqc_prefix, qc_config=variant_qc_config)
 
     # 2. Split into ethnic datasets and then QC each local dataset
-    logger.info("Splitting ethnic dataset")
-    nodes = SplitTGHeter().split(input_prefix=varqc_prefix, make_pgen=True)
+    nodes = set(TG_SUPERPOP_DICT.values())
```

Comment:
Receive `nodes` list from `SplitHeter` class instead of `TG_SUPERPOP_DICT`, because it won't work for homogeneous split.
```suggestion
    
```